### ISSUE TITLE
use count in template_file resources

### ISSUE THREAD

**Issue by rimusz** [hashicorp/terraform#2155]
Would be nice to have an option `template_file` resources to use count.
e.g. concat a few variables into a single string and reuse it in another resource

**Comment by phinze** [comment]
Hi @rimusz - can you provide an example of what you mean?

The title of your issue made it sound like #1893 documents what you're looking for, but "concat a few variables into a single string" sound like the `format()` [interpolation](https://terraform.io/docs/configuration/interpolation.html) function.

**Comment by rimusz** [comment]
@phinze this is my post in google group https://groups.google.com/forum/#!topic/terraform-tool/0UTDy_8rSDQ

It explains what I'm looking for.

**Comment by phinze** [comment]
Ah right - that post was close to the stop of my stack, but I didn't realize you were the same person. :)

Grabbing the salient parts of that post...

Given:

```js
variable "static_ip" {
    default = {
      "1" = "10.10.1.1"
      "2" = "10.10.2.1"
      "3" = "10.10.3.1"
    }
}
```

How to concat in the format like this:

```
etcd1=http://10.10.1.1:2380,etcd2=http://10.10.2.1:2380,etcd3=http://10.10.3.1:2380
```

I think to accomplish this we need #1915 to be done, after which this would be possible:

```js
// doesn't work yet, depends on #1915
output "combined" {
  value = "$(join(",", formatlist("etcd%s=%s", keys(var.static_ip), values(var.static_ip)))}"
}
```

**Comment by rimusz** [comment]
@phinze it sounds exactly what I'm looking for.
When we can expect that merged and released?

**Comment by phinze** [comment]
@rimusz done in #1915 - merged in master, will be released in 0.6 :grinning:

**Comment by rimusz** [comment]
@phinze awesome, thanks
